src/Documenter.jl

```diff
@@ -599,15 +599,15 @@ and when building docs for a tag they are deployed to a `vX.Y.Z` directory.
 function git_push(
         root, temp, repo;
         branch="gh-pages", dirname="", target="site", tag="", key="", sha="", devurl="dev",
-        versions, forcepush=false,
+        versions, forcepush=false, host="github.com",
```

Comment:
Are you planning to call `git_push` directly or where is this kwarg set? Couldn't we find the host from the `repo` argument to `deploydocs`?

Reply:
Yes currently I call `git_push` directly as in the first comment. We could definitely pull the host from the `repo` argument as something like `split(repo, "/")[1]` if you think that would be better - it's probably more robust.

Reply:
Yeah, parsing `repo` seems better here.

I think currently we allow both `github.com:` and `github.com/` as the start of the `repo`. So we could formalize that a bit and say repo has to start with the domain name, then you have `/` or `:` as a separator, and then the rest of the URL follows. So we should also split by `:`, or probably use a regex to match the start.